torch/serialization.py

```diff
@@ -318,6 +321,16 @@ def restore_location(storage, location):
     elif isinstance(map_location, _string_classes):
         def restore_location(storage, location):
             return default_restore_location(storage, map_location)
+    elif isinstance(map_location, torch.device):
+        if map_location.type == 'cpu':
+            map_str = 'cpu'
+        elif map_location.type == 'cuda':
+            map_str = 'cuda:{}'.format(map_location.index or 0)
+        else:
+            raise ValueError("The given map_location device is not a cpu or cuda")
```

Comment:
I think we can safely change all this code to `map_str = str(map_location)`

Reply:
Done